src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestGenieTelemetryService.kt

```diff
@@ -43,25 +50,29 @@ class TestGenieTelemetryService(_project: Project) {
             return
         }
 
-        val testCasesToSubmit = mutableListOf<ModifiedTestCase>()
+        val rawTestCasesToSubmit = mutableListOf<ModifiedTestCase>()
 
         synchronized(modifiedTestCasesLock) {
-            testCasesToSubmit.addAll(modifiedTestCases)
+            rawTestCasesToSubmit.addAll(modifiedTestCases)
             modifiedTestCases.clear()
         }
 
         // If there are no tests to submit, do not create a file
-        if (testCasesToSubmit.size == 0) {
+        if (rawTestCasesToSubmit.size == 0) {
             return
         }
 
-        log.info("Submiting ${testCasesToSubmit.size} test cases to a file")
+        ApplicationManager.getApplication().runReadAction {
+            val testCasesToSubmit = rawTestCasesToSubmit.map { it.convertToModifiedTestCaseWithAssertions(project) }
```

Comment:
what happens here?

Reply:
This converts the `ModifiedTestCase`s into `ModifiedTestCaseWithAssertions`

Reply:
oooooooooooh right